python/knext/knext/operator/spg_record.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Ant Group CO., Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied.
+
+import pprint
+from typing import Dict, Any, List, Tuple
+from knext.common.schema_helper import (
+    SPGTypeName,
+    PropertyName,
+    RelationName,
+)
+
+
+class SPGRecord:
+    """Data structure in operator, used to store entity information."""
+
+    def __init__(self, spg_type_name: SPGTypeName):
+        self._spg_type_name = spg_type_name
+        self._properties = {}
+        self._relations = {}
+
+    @property
+    def spg_type_name(self) -> SPGTypeName:
+        """Gets the spg_type_name of this SPGRecord.  # noqa: E501
+
+
+        :return: The spg_type_name of this SPGRecord.  # noqa: E501
+        :rtype: str
+        """
+        return self._spg_type_name
+
+    @spg_type_name.setter
+    def spg_type_name(self, spg_type_name: SPGTypeName):
+        """Sets the spg_type_name of this SPGRecord.
+
+
+        :param spg_type_name: The spg_type_name of this SPGRecord.  # noqa: E501
+        :type: str
+        """
+        self._spg_type_name = spg_type_name
+
+    @property
+    def properties(self) -> Dict[PropertyName, str]:
+        """Gets the properties of this SPGRecord.  # noqa: E501
+
+
+        :return: The properties of this SPGRecord.  # noqa: E501
+        :rtype: dict
+        """
+        return self._properties
+
+    @properties.setter
+    def properties(self, properties: Dict[PropertyName, str]):
+        """Sets the properties of this SPGRecord.
+
+
+        :param properties: The properties of this SPGRecord.  # noqa: E501
+        :type: dict
+        """
+        self._properties = properties
```

Comment:
add verification logic to determine whether each property in `Dict` belongs to the specified `_spg_type_name` SPGType.